crypto/rand/rand_lib.c

```diff
@@ -876,6 +882,9 @@ int rand_bytes_ex(OPENSSL_CTX *ctx, unsigned char *buf, int num)
     int ret;
     const RAND_METHOD *meth = RAND_get_rand_method();
 
+    if (meth == NULL)
```

Comment:
@slontis found the code path...

This check here is wrong as is, it needs to be removed.  Then add a `meth != NULL &&` to the test against `RAND_OpenSSL()` below.

If the FIPS case, there is a second copy of this code running in the provider and both `RAND_get_rand_method()` and `RAND_OpenSSL` legitimately return NULL.  The provider relies on both `rand_bytes` functions falling through to the drbg case below.  By returning error instead it fails.

I'm assuming that's the only problem introduced.  😸